desktop-app/app/reducers/browser.js

```diff
@@ -517,6 +521,28 @@ export default function browser(
           proxy: action.profile,
         },
       };
+    case TOGGLE_ALL_DEVICES_DESIGN_MODE:
+      const nextDevices = state.devices;
+      nextDevices.forEach(d => (d.designMode = action.allDevicesInDesignMode));
+      return {
+        ...state,
+        allDevicesInDesignMode: action.allDevicesInDesignMode,
+        devices: nextDevices,
+      };
+    case TOGGLE_DEVICE_DESIGN_MODE:
+      const deviceIndex = state.devices.findIndex(
```

Comment:
I think we should do, `nextDevices = [state.devices]` to avoid modifying the state.

Reply:
see my previous comment